src/vs/workbench/contrib/notebook/browser/contrib/coreActions.ts

```diff
@@ -1308,6 +1308,11 @@ registerAction2(class EditCellAction extends NotebookCellAction {
 	}
 
 	async runWithContext(accessor: ServicesAccessor, context: INotebookCellActionContext): Promise<void> {
+		const viewModel = context.notebookEditor.viewModel;
```

Comment:
We can add `NOTEBOOK_EDITOR_EDITABLE.isEqualTo(true)` to the `keybinding` when clause too, ensure the command is not triggered at all.